### PR TITLE
Default wallet setting to "Brave Wallet" causes some webcompat issues

### PR DESCRIPTION
**Pull request by bbondy**
The problem is that the ethereum object was set to non writable too early.
This moves the code to create the needed proxy to the native side.

Recall that code for deleting a property returns true was needed because of this:
https://github.com/brave/brave-core/commit/819279a8016264f2299085632d91d4c2b3912a2b

Resolves https://github.com/brave/brave-browser/issues/23546

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Try to login / connect with release channel (1.39) it will fail https://wallet.polygon.technology/
- Try the same steps with the fixed version, it will ask you to sign then log you in.

### CODE REVIEW

**Comment by bbondy** [comment]
CI failures are related to:
BatAdsNewTabPageAdIfAdsDisabledTest.FireEventIfNotExceededAdsPerHourCap and BatAdsNewTabPageServingIntegrationTest.ServeAdIfNotExceededAdsPerHourPermissionRuleFrequencyCap 
Fixed in: https://github.com/brave/brave-core/pull/13846

**Comment by kjozwiak** [comment]
I didn't have any issues signing in via `1.39.122 Chromium: 102.0.5005.115`. Basically went through the following:

* launched with a new `1.39.122 Chromium: 102.0.5005.115` profile on `Win 11 x64`
* created a wallet using `brave://wallet`
* ensured `Brave Wallet` was selected under `Default cryptocurrency wallet` via `brave://settings/wallet`
* visited https://wallet.polygon.technology and logged in after signing without issues

image

Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.42.6 Chromium: 103.0.5060.53 (Official Build) nightly (64-bit)
-- | --
Revision | a1711811edd74ff1cf2150f36ffa3b0dae40b17f-refs/branch-heads/5060@{#853}
OS | Windows 11 Version 21H2 (Build 22000.739)
```

Basically went through the same STR/Cases mentioned above which were derived from the STR/Cases via https://github.com/brave/brave-core/pull/13854#issue-1275427659.

**`Example`** | **`Example`** | **`Example`** | **`Example`**
---------------|-----------------|----------------|-----------------
 |  |  |
